src/PKSim.Core/Snapshots/Mappers/MoleculeMapper.cs

```diff
@@ -78,14 +82,16 @@ private void updateMoleculePropertiesToMolecule(IndividualMolecule molecule, Mol
          }
       }
 
-      public override async Task<IndividualMolecule> MapToModel(Molecule snapshot, ISimulationSubject simulationSubject)
+      public override async Task<IndividualMolecule> MapToModel(Molecule snapshot, ModelIndividual individual)
       {
-         var molecule = createMoleculeFrom(snapshot, simulationSubject);
+         var molecule = createMoleculeFrom(snapshot, individual);
          await UpdateParametersFromSnapshot(snapshot, molecule, snapshot.Type.ToString());
          MapSnapshotPropertiesToModel(snapshot, molecule);
          updateMoleculePropertiesToMolecule(molecule, snapshot);
-         molecule.Ontogeny = await _ontogenyMapper.MapToModel(snapshot.Ontogeny, simulationSubject);
-         await updateExpression(snapshot, new ExpressionContainerMapperContext {Molecule = molecule, SimulationSubject = simulationSubject});
+         var ontogeny = await _ontogenyMapper.MapToModel(snapshot.Ontogeny, individual);
+         _ontogenyTask.SetOntogenyForMolecule(molecule, ontogeny, individual);
```

Comment:
use the OntogenyTask instead of setting the ontogeny directly in the molecule. This will update all age dependent ontogeny